nutils/log.py

```diff
@@ -487,27 +478,44 @@ def range( title, *args ):
   '''Progress logger identical to built in range'''
 
   items = _range( *args )
-  return _logiter( title, _iter(items), length=len(items), useitem=True )
+  log = _getlog()
+  for item in items:
+    with log.context( '{} {} ({:.0f}%)'.format( title, item, (item+.5) * 100. / len(items) ) ):
+      yield item
```

Comment:
Does it make sense to add `50 / len(items)` to the percentage? The work for this iteration is done *after* logging the context.